libra_node/src/main_node.rs

```diff
@@ -244,6 +244,7 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> (AdmissionControlClien
         state_sync_network_sender,
         state_sync_network_events,
         &node_config,
+        vec![], // TODO: pass in empty vector for now, will be derived from node config later
```

Comment:
Since network also has a notion of trusted peers (see `network/src/protocols/discovery/mod.rs`), I'm wondering if we should just share access to this struct. As part of network construction, we today pass in a `HashMap<PeerId, NetworkPublicKeys>` as trusted_peer. Instead, we could construct an `Arc<RwLock<HashMap<...>>>` in `main_node.rs` and pass it to both state sync and network to share access.

Reply:
Why do you want to do it?
StateSynchronizer doesn't need NetworkPublicKeys